core/src/main/java/bisq/core/offer/takeoffer/TakeOfferModel.java

```diff
@@ -141,12 +160,19 @@ private void calculateTxFees() {
         // Payout tx: 371 bytes
         // Disputed payout tx: 408 bytes
 
-        // Request actual fees:
-        log.info("Start requestTxFee: txFeeFromFeeService={}", txFeeFromFeeService);
-        feeService.requestFees(() -> {
-            txFeePerByteFromFeeService = feeService.getTxFeePerByte();
-            txFeeFromFeeService = offerUtil.getTxFeeBySize(txFeePerByteFromFeeService, feeTxSize);
-        });
+        txFeePerByteFromFeeService = getTxFeePerByte();
+        txFeeFromFeeService = offerUtil.getTxFeeBySize(txFeePerByteFromFeeService, feeTxSize);
+        log.info("{} txFeePerByte = {}", feeService.getClass().getSimpleName(), txFeePerByteFromFeeService);
+    }
+
+    private Coin getTxFeePerByte() {
+        try {
+            CompletableFuture<Void> feeRequestFuture = CompletableFuture.runAsync(feeService::requestFees);
+            feeRequestFuture.get();  // Block until async fee request is complete.
```

Comment:
My bad, thought this was already used by UI. Would be better to handle this with callbacks. That way it should be usable by both API and UI eventually.

Reply:
If I understand correctly (@sqrrm), you want this `TakeOfferModel` snippet changed to using a callback instead of a `CompletableFuture`.

Do you mind if I make this change in a later PR?  I am working on a chain of 15 pr/branches, and this is branch #1 in the chain.

Reply:
Yeah, can be done in a later PR. Having it block is still more correct than before and since this is only used by API for now, it's not a problem to merge.